### PR TITLE
Add `os` module for access to environment variables

### PR DESCRIPTION
**Pull request by J3RN**
Does exactly what it says on the tin.

The function names originally ended in `_env`, but I figured `env.get_env(...)` was a bit redundant, so the `_env` suffixes have been removed.

### CODE REVIEW

File: src/gleam_erlang_ffi.erl [hunk 1]
```diff
@@ -109,3 +110,24 @@ delete_directory(Dir) ->
 
 recursive_delete(Dir) ->
     posix_result(file:del_dir_r(Dir)).
+
+get_all_env() ->
+    BinVars = lists:map(fun(VarString) ->
+                                [VarName, VarVal] = string:split(VarString, "="),
+                                {list_to_binary(VarName), list_to_binary(VarVal)}
+                        end, os:getenv()),
+    maps:from_list(BinVars).
```
Comment (lpil) [inline src/gleam_erlang_ffi.erl:119]:
```suggestion
    BinVars = lists:map(fun(VarString) ->
        [VarName, VarVal] = string:split(VarString, "="),
        {list_to_binary(VarName), list_to_binary(VarVal)}
    end, os:getenv()),
    maps:from_list(BinVars).
```

**Review by lpil — COMMENTED** [review]
Little nitpick: Should we put this in an `os` module rather than an `env`? It could also have other functions similar to the Erlang and Python `os` modules.

File: src/gleam_erlang_ffi.erl (hunk 1 above)
Comment (J3RN) [reply src/gleam_erlang_ffi.erl:119]: Yeah, sorry about this. Erlang ships with some ELisp the provide formatting, syntax highlighting, etc for Emacs. This is how that ELisp wanted this formatted :laughing: I meant to leave a comment about how off-the-wall crazy it is.

**Comment by J3RN** [comment]
I did ponder writing an `os` module instead, but I was worried about the cohesiveness of such a module; e.g. Elixir's `System` is quite a jumble. That said, Erlang's `os` is more focused, so as long as we commit to that paradigm I'll happily rename `env` to `os`. :grin:

File: test/gleam/erlang/env_test.gleam [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  assert Error(Nil) = os.get("MYVAR")
+
+  os.set("MYVAR", "MYVALUE")
+
+  assert Nil = os.unset(name: "MYVAR")
+  assert Error(Nil) = os.get(name: "MYVAR")
+}
+
+pub fn get_non_existant_test() {
+  // Just to make sure
```
Comment (lpil) [inline test/gleam/erlang/env_test.gleam:36]: 😁

**Review by lpil — APPROVED** [review]
Wonderful! Thank you again

**Pull request merged**